esrally/async_connection.py

```diff
@@ -16,6 +86,65 @@ async def text(self, encoding=None, errors="strict"):
         return self._body
 
 
+class ResponseMatcher:
+    def __init__(self, responses):
+        self.logger = logging.getLogger(__name__)
+        self.responses = []
+
+        for response in responses:
+            path = response["path"]
+            if path == "*":
+                matcher = ResponseMatcher.always()
+            elif path.startswith("*"):
+                matcher = ResponseMatcher.endswith(path[1:])
+            elif path.endswith("*"):
+                matcher = ResponseMatcher.startswith(path[:-1])
+            else:
+                matcher = ResponseMatcher.equals(path)
+
+            body = response["body"]
+            body_encoding = response.get("body-encoding", "json")
+            if body_encoding == "raw":
+                body = json.dumps(body).encode("utf-8")
+            elif body_encoding == "json":
+                body = json.dumps(body)
+            else:
+                raise ValueError(f"Unknown body encoding [{body_encoding}] for path [{path}]")
+
+            self.responses.append((path, matcher, body))
+
+    @staticmethod
+    def always():
+        # pylint: disable=unused-variable
+        def f(p):
+            return True
+        return f
```

Comment:
alternatively we could replace the closures with more Pythonic (IMHO? :) ) lambdas e.g.

`return lambda f: True`

or for `startswith`:
```
def startswith(path_pattern):
    return lambda f: f.startswith(path_pattern)
```

Reply:
I implemented this with lambdas earlier but then got a PEP-8 violation warning. :)

Reply:
LOL whaaaaaat?